packages/core/src/types/token.ts

```diff
@@ -1,22 +1,71 @@
+import invariant from 'tiny-invariant'
 import { type Address, isHex, zeroAddress } from 'viem'
-import { tokenMapping } from '../constants.js'
+import type { Exchange } from './wallet.js'
+
+////////////////////////////////////////////////////////////////////////////////
+// Constants
+////////////////////////////////////////////////////////////////////////////////
+
+export const STABLECOINS = ['USDC', 'USDT']
+
+////////////////////////////////////////////////////////////////////////////////
+// Token Mapping
+////////////////////////////////////////////////////////////////////////////////
+
+type SupportedExchange = 'Binance' | 'Coinbase' | 'Kraken' | 'Okx'
+
+type TokenMetadata = {
+  name: string
+  ticker: string
+  address: Address
+  decimals: number
+  supported_exchanges: Partial<Record<SupportedExchange, string>>
+  chain_addresses: Record<string, string>
+  logo_url: string
+}
+
+type TokenMapping = {
+  tokens: TokenMetadata[]
+}
+
+const tokenMappingUrl =
+  process.env.TOKEN_MAPPING_URL || process.env.NEXT_PUBIC_TOKEN_MAPPING_URL
+
+const tokenMappingStr =
+  process.env.NEXT_PUBLIC_TOKEN_MAPPING || process.env.TOKEN_MAPPING
+
+invariant(
+  tokenMappingUrl || tokenMappingStr,
+  'No token mapping initialization option provided',
+)
+
+export const tokenMapping = JSON.parse(
+  tokenMappingUrl
+    ? await fetch(tokenMappingUrl).then((res) => res.json())
```

Comment:
@sehyunc is this top-level `await` ok?

Reply:
Ok in the frontend, TBD in the bot server